README.md

```diff
@@ -259,6 +259,46 @@ Additional configuration are created in /etc/nginx/conf.d/
                - root "/usr/share/nginx/html"
                - index index.html
 ```
+8) Site configuration using a custom template.
+Instead of defining a site config file using a list of attributes,
+you may use a hash that indicates an alternate template.
+Additional hash elements are accessible inside the template through
+the `nginx_sites[item]` hash.
+```yaml
+- hosts: all
+
+  roles:
+  - role: nginx
+    nginx_sites:
+      bar:
+        template: bar.conf.j2
+        server_name: bar.example.com
```

Comment:
Thanks for the quick review.

I agree.  The stinky part of this solution is that nginx_sites[item] is intended to be a list all through the implementation (and documentation,) and this does expect a hash.  After digging through the source, the only place that matters is in the site.conf.j2 template (which is not used when providing a custom template, and which is why this works.)  But I agree, it's "tricky" and  that usually ends up causing trouble down the road.

Using a list of string directives does not give many options  besides scanning the strings content.  The idea to 
> detect if there is "server {" inside

is an interesting one.  I suppose the "server { ... }" wrapper could be _optionally_ excluded when the list of directives includes its own server block.  What I don't like about that is that this makes the site.conf template a little more magical and "dsl-like."  But this could work well, is easily explained, and would allow virtually any configuration.

The only other option I considered was to introduce a new hash of site definitions similar to the `nginx_sites`, perhaps named `templated_nginx_sites` (though hopefully, something less clumsy) that would work just like the nginx_sites, but would expect a template and optional template args.  But providing two separate ways to create sites has its own issues.

This PR was for an immediate requirement and we're running off the flatrocks fork for now, but I am interested in a more permanent solution, and glad to pitch in if it would be helpful.